src/main/shell-session.ts

```diff
@@ -105,13 +105,14 @@ export class ShellSession extends EventEmitter {
   protected async getShellEnv() {
     const env = JSON.parse(JSON.stringify(await shellEnv()));
     const pathStr = [this.kubectlBinDir, this.helmBinDir, process.env.PATH].join(path.delimiter);
+    const shell = userStore.preferences.shell && userStore.preferences.shell !== "" ? userStore.preferences.shell : process.env.SHELL;
 
     if(isWindows) {
       env["SystemRoot"] = process.env.SystemRoot;
-      env["PTYSHELL"] = process.env.SHELL || "powershell.exe";
+      env["PTYSHELL"] = shell || "powershell.exe";
```

Comment:
Maybe we should also respect the user's `PTYSHELL` setting?
```suggestion
      env["PTYSHELL"] ||= shell || "powershell.exe";
```

Reply:
that would prevent overriding the shell

Reply:
How so? Or do you mean that if this setting is set we want to override `PTYSHELL` no matter what?

The solution then would be:

```suggestion
      env["PTYSHELL"] = shell || env["PTYSHELL"] || "powershell.exe";
```

Reply:
Yes, that's the purpose.

Reply:
`PTYSHELL` shell is now taken into an account earlier.